Mathlib/CategoryTheory/Shift/Twist.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+module
+
+public import Mathlib.CategoryTheory.Center.Basic
+public import Mathlib.CategoryTheory.Shift.CommShift
+
+/-!
+# Twisting a shift
+
+Given a category `C` equipped with a shift by a monoid `A`, we introduce a structure
+`t : TwistShiftData C A` which consists of a collection of invertible elements in the
+center of the category `C` (typically, this will be suitable signs when we
+assume `C` is preadditive), which allow to introduce a type synonym
+category `TwistShift t` which identical shift functors as `C` but where
+the isomorphisms `shiftFunctorAdd` have been modified.
+
+-/
+
+@[expose] public section
+
+universe w v u
+
+namespace CategoryTheory
+
+variable (C : Type u) [Category.{v} C] (A : Type w) [AddMonoid A] [HasShift C A]
+
+/-- Given a category `C` equipped with a shift by a monoid `A` -/
+structure TwistShiftData where
+  /-- The invertible elements in the center of `C` which are used to
+  modify the `shiftFunctorAdd` isomorphisms . -/
+  z (a b : A) : (CatCenter C)ˣ
+  z_zero_right (a : A) : z a 0 = 1 := by cat_disch
+  z_zero_left (b : A) : z 0 b = 1 := by cat_disch
+  assoc (a b c : A) : z (a + b) c * z a b = z a (b + c) * z b c := by cat_disch
```

Comment:
You only need `z 0 0 = 1` (instead of `z_zero_left` and `z_zero_right`) because `assoc (a := 0) (b := 0)` tells you `z 0 0 = z 0 c` etc.